MetaMask/metamask-extension
# Add snaps name provider

## **Description**

Add the `SnapsNameProvider` to use installed snaps with the `name-lookup` endowment to propose names for Ethereum addresses in the `Name` component.

## **Manual Testing Steps**

1. Go to test snaps.
2. Install the `Name Lookup` snap.
3. Go to test dApp.
4. Create a `V3` or `V4` typed signature request.
5. Click an address.
6. Verify the proposed names include a `Name Lookup Example Snap` option.

## **Screenshots**

<img width="359" alt="snaps_provider" src="https://github.com/MetaMask/metamask-extension/assets/19745142/0ae7d840-7e53-4972-8441-0a504871bf88">

## **Related Issues**

Fixes [#1235](https://github.com/MetaMask/MetaMask-planning/issues/1235)

## **Pre-merge Author Checklist**


## **Pre-merge Reviewer Checklist**
